crates/wallet/src/wallet/persisted.rs

```diff
@@ -120,7 +120,7 @@ pub trait AsyncWalletPersister {
 #[derive(Debug)]
 pub struct PersistedWallet<P> {
     inner: Wallet,
-    marker: PhantomData<P>,
+    _marker: PhantomData<fn(P) -> P>,
```

Comment:
```suggestion
    _marker: PhantomData<*const P>,
```

According to @LLFourn

Reply:
I tried this approach and get this error running the `test_thread_safety` test:
```
error[E0277]: `*const Connection` cannot be sent between threads safely
    --> crates/wallet/tests/wallet.rs:4215:19
     |
4215 |     thread_safe::<PersistedWallet<bdk_chain::rusqlite::Connection>>();
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `*const Connection` cannot be sent between threads safely
     |
     = help: within `PersistedWallet<Connection>`, the trait `Send` is not implemented for `*const Connection`
     = help: the trait `Send` is implemented for `Connection`
note: required because it appears within the type `PhantomData<*const Connection>`
    --> /Users/steve/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/marker.rs:757:12
     |
757  | pub struct PhantomData<T: ?Sized>;
     |            ^^^^^^^^^^^
note: required because it appears within the type `PersistedWallet<Connection>`
    --> /Users/steve/git/notmandatory/bdk/crates/wallet/src/wallet/persisted.rs:121:12
     |
121  | pub struct PersistedWallet<P> {
     |            ^^^^^^^^^^^^^^^
note: required by a bound in `thread_safe`
    --> crates/wallet/tests/wallet.rs:4212:23
     |
4212 |     fn thread_safe<T: Send + Sync>() {}
     |                       ^^^^ required by this bound in `thread_safe`
```

Reply:
`PhantomData<*const P>` makes it `!Sync`: https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns

Reply:
@notmandatory Sorry yeah my suggestion does the complete opposite of what we need.